redis/cluster.py

```diff
@@ -425,27 +426,30 @@ class initializer. In the case of conflicting arguments, querystring
 
     def __init__(
         self,
-        host=None,
-        port=6379,
-        startup_nodes=None,
-        cluster_error_retry_attempts=3,
-        require_full_coverage=False,
-        reinitialize_steps=10,
-        read_from_replicas=False,
-        dynamic_startup_nodes=True,
-        url=None,
+        host: Optional[str] = None,
+        port: int = 6379,
+        startup_nodes: Optional[List["ClusterNode"]] = None,
+        cluster_error_retry_attempts: int = 3,
+        connection_error_retry_attempts: int = 3,
+        retry: Optional["Retry"] = None,
```

Comment:
What do you think about collapsing retry and retry_on_error?

Reply:
We can remove retry_on_error all together, as the Retry object can be set with specific errors (supported_errors) and it defaults to ConnectionError and TimeoutError. The only issue with removing the retry_on_error is backward compatibility.

Reply:
I'd rather do this removal in redis-py <next major version> So.. yuck, but let's leave it.